waltz-tools/src/main/java/com/wepay/waltz/tools/cluster/ClusterCli.java

```diff
@@ -767,7 +809,8 @@ protected String getUsage() {
 
             PartitionValidationResults(int partitionId) {
                 this.partitionId = partitionId;
-                this.validationResultsMap = new EnumMap<>(ValidationResult.ValidationType.class);
+                this.validationResultsMap =
+                    Collections.synchronizedMap(new EnumMap<>(ValidationResult.ValidationType.class));
```

Comment:
synchronizedMap makes sure threads have an up-to-date view of all buckets plus allows null values. Is that why we chose it? It's probably not a big deal but I am curious why ConcurrentHashMap was not of choice here.

Reply:
Because I wanted to keep the map as a EnumMap. As we have a synchronizedMap that we can wrap around any other Map, I thought this can be used. 

If we are not very particular to use EnumMap then we can instead change to ConcurrentHashMap.

Reply:
Ow I missed that... Yeah... Forget what I said, we won't have that many threads modifying a partition map. We have only 5 keys in our enum :)